addons/web/static/tests/core/dropdown_tests.js

```diff
@@ -1313,4 +1315,63 @@ QUnit.module("Components", ({ beforeEach }) => {
             '<div class="o-dropdown dropdown o-dropdown--no-caret"><button class="dropdown-toggle" disabled="" tabindex="0" aria-expanded="false"></button></div>'
         );
     });
+
+    QUnit.test("don't close dropdown outside the active element", async (assert) => {
+        // This test checks that if a dropdown element opens a dialog with a dropdown inside,
+        // opening this dropdown will not close the first dropdown.
+        patchWithCleanup(Dropdown.prototype, {
+            close() {
+                assert.notOk(true, "dropdown should not be closed");
+                this._super();
+            },
+        });
+
+        class CustomDialog extends Component {}
+        CustomDialog.template = xml`
+            <Dialog title="'Welcome'">
+                <Dropdown>
+                    <DropdownItem>Item</DropdownItem>
+                </Dropdown>
+            </Dialog>`;
+        CustomDialog.components = { Dialog, Dropdown, DropdownItem };
+
+        const mainComponentRegistry = registry.category("main_components");
+        clearRegistryWithCleanup(mainComponentRegistry);
+        serviceRegistry.add("dialog", dialogService);
+        serviceRegistry.add("l10n", makeFakeLocalizationService());
```

Comment:
why do we need this?

Reply:
Because, one of the components on the mainComponentRegistry is "BlockUI" and it uses this.env._t in his code.